WebXR build fails on visionOS

mwyrzykowski opened the pull request:
#### 543e3c1db166e5ee704c71ca609ff1a376a0e6bc
<pre>
WebXR build fails on visionOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=259535">https://bugs.webkit.org/show_bug.cgi?id=259535</a>
&lt;radar://112938443&gt;

Reviewed by Matt Woodrow and Dean Jackson.

Fix build for visionOS after 266327@main

* Source/WebCore/Modules/webxr/WebXROpaqueFramebuffer.cpp:
(WebCore::makeEGLImageSource):
* Source/WebCore/platform/xr/PlatformXR.h:
(PlatformXR::Device::FrameData::LayerData::encode const):
* Source/WebKit/Shared/RemoteLayerTree/CGDisplayList.cpp:
(WebKit::CGDisplayList::encode):
(WebKit::CGDisplayList::encode const): Deleted.
* Source/WebKit/Shared/RemoteLayerTree/CGDisplayList.h:

Canonical link: <a href="https://commits.webkit.org/266347@main">https://commits.webkit.org/266347@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/57b6aab82dd13da09659763ba9f9b0ca4872ce51

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/13586 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/13900 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/14233 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/15322 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/12906 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/13667 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/16408 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/13980 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/15595 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/13752 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/14391 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/11504 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/16022 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/11679 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/12259 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/19295 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/12754 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/12427 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/15632 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/12938 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/10827 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/12207 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3311 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/16537 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/12781 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mwyrzykowski commented on Source/WebKit/Shared/RemoteLayerTree/CGDisplayList.cpp:
```diff
@@ -32,10 +32,10 @@
 {
     encoder << m_displayList;
-    encoder << m_surfaces;
+    encoder << WTFMove(m_surfaces);
```
@hortont424 is it correct to move `m_surfaces` here? I tried it, it seems to work, but wanted to check with you prior to merging

rr-codes replied:
Encoding something shouldn't mutate its state, right?

mattwoodrow replied:
I think it's correct in this case, the type of `this` is an r-value reference, so the caller has promised us that the object won't be used again, and we can consume its contents as we desire.

Ultimately the innermost type here (`MachSendRight`) is actually consumed by writing it to IPC (the kernel transfers ownership of the object to the new process), so propagating that up into wrapping types makes sense (or they can make an explicit copy of MachSendRights if they want).

mwyrzykowski committed:
WebXR build fails on visionOS
https://bugs.webkit.org/show_bug.cgi?id=259535
<radar://112938443>

Reviewed by Matt Woodrow and Dean Jackson.

Fix build for visionOS after 266327@main

* Source/WebCore/Modules/webxr/WebXROpaqueFramebuffer.cpp:
(WebCore::makeEGLImageSource):
* Source/WebCore/platform/xr/PlatformXR.h:
(PlatformXR::Device::FrameData::LayerData::encode const):
* Source/WebKit/Shared/RemoteLayerTree/CGDisplayList.cpp:
(WebKit::CGDisplayList::encode):
(WebKit::CGDisplayList::encode const): Deleted.
* Source/WebKit/Shared/RemoteLayerTree/CGDisplayList.h:

Canonical link: https://commits.webkit.org/266347@main
